src/west/_bootstrap/main.py

```diff
@@ -231,16 +221,37 @@ def bootstrap(args):
     # Clone the west source code and the manifest into west/. Git will create
     # the west/ directory if it does not exist.
 
-    clone('west repository', west_url, west_rev,
-          os.path.join(directory, WEST_DIR, WEST))
-
     clone('manifest repository', manifest_url, manifest_rev,
           os.path.join(directory, WEST_DIR, MANIFEST))
 
+    # Parse the manifest and look for a section named "west"
+    manifest_file = os.path.join(directory, WEST_DIR, MANIFEST, 'default.yml')
+    with open(manifest_file, 'r') as f:
+        data = yaml.safe_load(f.read())
+
+    if 'west' in data:
+        wdata = data['west']
+        try:
+            pykwalify.core.Core(
+                source_data=wdata,
+                schema_files=[_SCHEMA_PATH]
+            ).validate()
+        except pykwalify.errors.SchemaError as e:
+            print('Error: Failed to parse manifest file: "{}."'.
+                  format(manifest_file), file=sys.stderr)
```

Comment:
If the exception from pykwalify has a useful message in it, then you could do this:

```python
sys.exit("Error: Failed to parse manifest file '{}': {}"
         .format(manifest_file, e))
```

`sys.exit()` in Python has a handy feature where it exits with 1 and prints a message if you pass it a string.

Reply:
That's much better, done